mozilla-release/mobile/android/base/java/org/mozilla/gecko/widget/themed/ThemedTabLayout.java

```diff
@@ -0,0 +1,188 @@
+import android.content.res.TypedArray;
+import android.graphics.drawable.ColorDrawable;
+import android.graphics.drawable.Drawable;
+import android.util.AttributeSet;
+
+import static org.mozilla.gecko.Tabs.registerOnTabsChangedListener;
+import static org.mozilla.gecko.Tabs.unregisterOnTabsChangedListener;
+
+public class ThemedTabLayout extends android.support.design.widget.TabLayout
+                                     implements LightweightTheme.OnChangeListener, Tabs.OnTabsChangedListener{
+    private LightweightTheme theme;
+
+    private static final int[] STATE_PRIVATE_MODE = { R.attr.state_private };
+    private static final int[] STATE_LIGHT = { R.attr.state_light };
+    private static final int[] STATE_DARK = { R.attr.state_dark };
+
+    protected static final int[] PRIVATE_PRESSED_STATE_SET = { R.attr.state_private, android.R.attr.state_pressed };
+    protected static final int[] PRIVATE_FOCUSED_STATE_SET = { R.attr.state_private, android.R.attr.state_focused };
+    protected static final int[] PRIVATE_STATE_SET = { R.attr.state_private };
+
+    private boolean isPrivate;
+    private boolean isLight;
+    private boolean isDark;
+    private boolean autoUpdateTheme;        // always false if there's no theme.
+
+    private ColorStateList drawableColors;
+
+    public ThemedTabLayout(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        initialize(context, attrs, 0);
+    }
+
+    public ThemedTabLayout(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        initialize(context, attrs, defStyle);
+    }
+
+    private void initialize(final Context context, final AttributeSet attrs, final int defStyle) {
+        // The theme can be null, particularly if we might be instantiating this
+        // View in an IDE, with no ambient GeckoApplication.
+        final Context applicationContext = context.getApplicationContext();
+        if (applicationContext instanceof GeckoApplication) {
+            theme = ((GeckoApplication) applicationContext).getLightweightTheme();
+        }
+
+        final TypedArray a = context.obtainStyledAttributes(attrs, R.styleable.LightweightTheme);
+        autoUpdateTheme = theme != null && a.getBoolean(R.styleable.LightweightTheme_autoUpdateTheme, true);
+        a.recycle();
+    }
+
+    @Override
+    protected void onAttachedToWindow() {
+        super.onAttachedToWindow();
+        registerOnTabsChangedListener(this);
+        if (autoUpdateTheme)
+            theme.addListener(this);
+    }
+
+    @Override
+    protected void onDetachedFromWindow() {
+        super.onDetachedFromWindow();
+        unregisterOnTabsChangedListener(this);
+        if (autoUpdateTheme)
+            theme.removeListener(this);
+    }
+
+    @Override
+    public int[] onCreateDrawableState(int extraSpace) {
+        final int[] drawableState = super.onCreateDrawableState(extraSpace + 1);
+
+        if (isPrivate)
```

Comment:
brackets

Reply:
@ravjit-cliqz, this file is generated by a script, I do not think that Moaz should start to correctly format generated files.